dbt/adapters/spark/relation.py

```diff
@@ -1,44 +1,23 @@
-from dbt.adapters.base.relation import BaseRelation
+from dataclasses import dataclass
 
+from dbt.adapters.base.relation import BaseRelation, Policy
+
+
+@dataclass
+class SparkQuotePolicy(Policy):
+    database: bool = False
+    schema: bool = False
+    identifier: bool = False
 
-class SparkRelation(BaseRelation):
-    DEFAULTS = {
-        'metadata': {
-            'type': 'SparkRelation'
-        },
-        'quote_character': '`',
-        'quote_policy': {
-            'database': False,
-            'schema': False,
-            'identifier': False,
-        },
-        'include_policy': {
-            'database': False,
-            'schema': True,
-            'identifier': True,
-        }
-    }
 
-    SCHEMA = {
-        'type': 'object',
-        'properties': {
-            'metadata': {
-                'type': 'object',
-                'properties': {
-                    'type': {
-                        'type': 'string',
-                        'const': 'SparkRelation',
-                    },
-                },
-            },
-            'type': {
-                'enum': BaseRelation.RelationTypes + [None]
-            },
-            'path': BaseRelation.PATH_SCHEMA,
-            'include_policy': BaseRelation.POLICY_SCHEMA,
-            'quote_policy': BaseRelation.POLICY_SCHEMA,
-            'quote_character': {'type': 'string'},
-        },
-        'required': ['metadata', 'type', 'path', 'include_policy',
-                     'quote_policy', 'quote_character']
-    }
+@dataclass
+class SparkIncludePolicy(Policy):
+    database: bool = False
+    schema: bool = True
+    identifier: bool = True
+
+
+@dataclass(frozen=True, eq=False, repr=False)
+class SparkRelation(BaseRelation):
+    quote_policy: SparkQuotePolicy = SparkQuotePolicy()
+    include_policy: SparkIncludePolicy = SparkIncludePolicy()
```

Comment:
Assuming this wasn't intended to change the quote character to `"`, I think you'll also want to include:
```python
quote_character: str = '`'
```

Reply:
I've actually overridden `SparkQuotePolicy` and set everything to not include quotes. I'm using this adapter with Databricks and there are restrictions on table names, specifically; 

> Valid names only contain alphabet characters, numbers and _.

I'm not sure if this is the same with Spark, but I can only assume. I'd like to just enforce no quotes if possible.

Reply:
It looks like Spark itself [encodes](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L114) that restriction on object name validity. I checked and saw that it does not enforce the same validation on column names, however.

There's a lot of dbt and dbt-adjacent functionality that uses the `quoted` versions of columns and relations by default, since it gives us more flexibility across databases. In general, even though quoting is far less important on Spark than on Redshift or Snowflake, I think it's a better idea to keep quoting implemented in this plugin, rather than needing to reimplement dozens of common macros.

A specific example that came up today: the [`dbt_utils.equality`](https://github.com/fishtown-analytics/dbt-utils/blob/master/macros/schema_tests/equality.sql#L12) test uses the `quoted` attribute from `get_columns_from_relation`, and missing that attribute caused an integration test to fail. I think our answer would be to override the `quoted` property of the `SparkColumn` class in the same way BigQuery does [here](https://github.com/fishtown-analytics/dbt/blob/dev/barbara-gittings/plugins/bigquery/dbt/adapters/bigquery/column.py#L77).

Reply:
Makes sense. I have implemented this change.